compilers/concrete-compiler/compiler/lib/Dialect/RT/Analysis/LowerDataflowTasksToRT.cpp

```diff
@@ -635,6 +635,24 @@ struct FixupBufferDeallocationPass
     for (auto op : ops) {
       op->erase();
     }
+
+    // For all task return ops, ensure that the return value is
```

Comment:
What does this prevent?

Reply:
This avoids having the return of a task be deallocated twice. The return is a future that gets deallocated once it is no longer needed, but that assumes that the contents of that future are allocated rather than aliased. In some tasks the return future is built using one of the input memrefs when doing in-place updates, but then that input itself can be deallocated and we can't decide this without inter-procedural analysis at the point where deallocation operations are introduced. So instead we make a copy when the return value is aliasing an input.